### ISSUE TITLE
Spherefun roots is broken

### ISSUE THREAD

**Issue by trefethen** [chebfun/chebfun#2157]
For some time now, Spherefun roots has been broken; unfortunately the associated test file is so minimal that it doesn't detect this.  Below is an example.

It is likely that the problem arose when @trefethen improved roots for Chebfun2 and SeparableApprox.  Nevertheless perhaps one of @ajt60gaibb, @gradywright, @heatherw3521 might be able to take a look and see what they think?

```
>> cheb.xyzsphere
>> roots(x-.2)
Error using spherefun/grad
Too many output arguments.

Error in separableApprox/roots (line 80)
        [fx, fy] = grad(f);

Error in spherefun/roots (line 25)
rts = roots@separableApprox(varargin{:});
```

**Comment by gradywright** [comment]
This is the same bug as #2118.

**Comment by ajt60gaibb** [comment]
This looks like a relatively serious bug because of the way vectors in `spherefunv` are represented in the cartesian coordinate system.  This could take a while...

**Comment by ajt60gaibb** [comment]
Partial fix merged in #2161.

**Comment by ajt60gaibb** [comment]
Duplicate of #2118 so closing.
